### PR TITLE
Add documentation and internal file too large test for ImageSeries

### LINKED ISSUE #258: [Add Check]: Add Best Practice for External Mode usage for ImageSeries
### What would you like to see added to the NWBInspector?

Came up in meeting today - add best practice section covering external mode usage for natural behavioral videos saved as ImageSeries.

### Pseudocode or other logic of the check

_No response_

### Do you have any interest in helping implement the check function?

Yes.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/.github/CODE_OF_CONDUCT.rst)
- [X] Have you ensured this check was not already [requested](https://github.com/neurodatawithoutborders/nwbinspector/issues)?

### PR DESCRIPTION
**Pull request by h-mayorquin**
Added the missing documentation in best practices and API for ImageSeries. Should close #258 when finished. 

@bendichter @rly 
Can any of you make me a contributor?

### CODE REVIEW

File: tests/unit_tests/test_image_series.py [hunk 1]
```diff
@@ -125,3 +127,26 @@ def test_check_image_series_external_file_relative(self):
 def test_check_image_series_external_file_valid_pass_non_external():
     image_series = ImageSeries(name="TestImageSeries", rate=1.0, data=np.zeros(shape=(3, 3, 3, 3)), unit="TestUnit")
     assert check_image_series_external_file_valid(image_series=image_series) is None
+
+
+def test_check_large_image_series_stored_internally():
```
Comment (CodyCBakerPhD) [inline tests/unit_tests/test_image_series.py:141]: We usually write two tests - one that raises a message and another that does not.

Can you write another that does not (that is, shows the 'correct' behavior)?

File: nwbinspector/checks/image_series.py [hunk 2]
```diff
@@ -38,3 +38,15 @@ def check_image_series_external_file_relative(image_series: ImageSeries):
...
+
+
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=ImageSeries)
+def check_image_series_too_large(image_series: ImageSeries, gb_lower_bound: float = 20.0):
+    """Check if an ImageSeries stored is larger than gb_lower_bound and suggests external file."""
+    data = image_series.data
+    data_size_gb = data.size * data.dtype.itemsize // 1e9
+    if data_size_gb > gb_lower_bound:
+        return InspectorMessage(
+            severity=Severity.HIGH,
```
Comment (CodyCBakerPhD) [inline nwbinspector/checks/image_series.py]: I'd only use `Severity` if you have two checks like this but where one is slightly more impactful than the other, such as the data compression checks: https://github.com/NeurodataWithoutBorders/nwbinspector/blob/dev/nwbinspector/checks/nwb_containers.py#L11-L58

File: tests/unit_tests/test_image_series.py [hunk 3]
```diff
@@ -125,3 +127,26 @@ def test_check_image_series_external_file_relative(self):
...
+
+    gb_size = 1.0
+    image_size = 10
+    total_elements = int(gb_size * 1e9 / np.dtype("float").itemsize) // (image_size * image_size)
```
Comment (CodyCBakerPhD) [inline tests/unit_tests/test_image_series.py]: Since you're already making use of the `gb_lower_bound` argument, I'd just go ahead and make this testing example somewhat smaller (does 10 MB sound reasonable?) so it doesn't slow down the tests too much.

File: docs/best_practices/image_series.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+
+When storing commmon multimedia formats using the :nwb-schema:ref:`sec-ImageSeries` the file should be stored as
+an external file. That is, the file should be packaged together with the nwb file instead of stored inside the format.
+This can be accomplished by using  the ``external_file`` file option to store the path instead.
+
+Check function: :py:meth:`~nwbinspector.checks.check_image_series_too_large`
+
+
+Use relative path for external mode
```
Comment (CodyCBakerPhD) [inline docs/best_practices/image_series.rst:21]: A really cool thing with our custom sphinx code here is you can form an API doc linkage with the BP section by defining a reference to this section of the following form: `.. _best_practice_<name_of_check_function_that_tests_it>:` or specific to this section `.. _best_practice_image_series_external_file_relative:`

Can you add these links here for both sections?

File: docs/best_practices/image_series.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+Storage of ImageSeries
+----------------------
+
+Use external model when file is too large
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+When storing commmon multimedia formats using the :nwb-schema:ref:`sec-ImageSeries` the file should be stored as
+an external file. That is, the file should be packaged together with the nwb file instead of stored inside the format.
+This can be accomplished by using  the ``external_file`` file option to store the path instead.
```
Comment (CodyCBakerPhD) [inline docs/best_practices/image_series.rst:15]: Both of the primary statements of these sections look fine to me.

In the BP sections we do generally try to attempt to explain the reasoning/justification for the practice - I'll illustrate that a bit in the issue https://github.com/NeurodataWithoutBorders/nwbinspector/issues/258 and then can you summarize it for each of these sections here?

File: tests/unit_tests/test_image_series.py (hunk 3 above)
Comment (h-mayorquin) [reply tests/unit_tests/test_image_series.py]: Sure.

File: nwbinspector/checks/image_series.py (hunk 2 above)
Comment (h-mayorquin) [reply nwbinspector/checks/image_series.py]: So, severity should be removed of the check? OK, done.

File: tests/unit_tests/test_image_series.py (hunk 1 above)
Comment (h-mayorquin) [reply tests/unit_tests/test_image_series.py:141]: All right, done.

File: docs/best_practices/image_series.rst (hunk 4 above)
Comment (h-mayorquin) [reply docs/best_practices/image_series.rst:21]: OK, see the ones that I add to see if I got you.

File: docs/best_practices/image_series.rst (hunk 5 above)
Comment (h-mayorquin) [reply docs/best_practices/image_series.rst:15]: I think you should write this. The explanation is not clear to me.

**Review by CodyCBakerPhD — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
LGTM, thanks for getting around to this.

File: docs/best_practices/image_series.rst (hunk 5 above)
Comment (CodyCBakerPhD) [reply docs/best_practices/image_series.rst:15]: I'll polish the docs myself later then

Comment (h-mayorquin) [reply docs/best_practices/image_series.rst:15]: Thanks @CodyCBakerPhD!
